src/towncrier/newsfragments/561.bugfix.rst

```diff
@@ -0,0 +1 @@
+Add explicit encoding to read_text.
```

Comment:
I don't know if anyone is using non UTF-8 encoding in these days for source code.... maybe some legacy Windows shops might still use UTF-16 

--------

I am ok with this change, but I think that somewhere in the documentation, we should mention that `towncrier` can only work with UTF-8 fragment files and it will produce UTF-8 encoded files.

Maybe this can be done as part of the "Philosophy" section

https://towncrier.readthedocs.io/en/latest/#philosophy

Maybe add 
```
It assumes that the source code is stored in UTF-8 files.
```

Reply:
> I don't know if anyone is using non UTF-8 encoding in these days for source code.... maybe some legacy Windows shops might still use UTF-16

In my experience, that's not been the case, even for text-oriented applications like towncrier. My instinct is it's still a bug fix, but for Windows users, it is technically a breaking change. My advice - hope that there aren't any Windows users relying on locale but be prepared to back out the change and introduce a compatibility shim if someone does encounter an issue. I'm happy to help with that if needed.

----



> Maybe this can be done as part of the "Philosophy" section

I'm a little uneasy adding it to the Philosophy section since that section comes from the readme, and it feels more like an implementation detail. Philosophy doesn't even mention markdown or restructured text, a concern that's going to be a lot more salient to a reader.

In 83b8c02, I've proposed that it be added to the tutorial, near where "markdown" is mentioned and shortly after the user has first encountered creating content for news fragments. Let me know what you think.

Reply:
In e1caa2c79d7219a6d41fd7d9f5efe30699da6145, I took it a step further and explicitly mentioned other concerns that expect UTF-8.